packages/node-experimental/src/sdk/scope.ts

```diff
@@ -49,18 +94,78 @@ export class OtelScope extends Scope {
    * @inheritDoc
    */
   protected _getBreadcrumbs(): Breadcrumb[] {
-    const transaction = getActiveTransaction();
-    const transactionBreadcrumbs = transaction && transaction.getBreadcrumbs ? transaction.getBreadcrumbs() : [];
+    const span = this.activeSpan || getActiveSpan();
 
-    return this._breadcrumbs.concat(transactionBreadcrumbs);
+    const spanBreadcrumbs = span ? getBreadcrumbsForSpan(span) : [];
+
+    return spanBreadcrumbs.length > 0 ? this._breadcrumbs.concat(spanBreadcrumbs) : this._breadcrumbs;
   }
 }
 
 /**
- * This gets the currently active transaction,
- * and ensures to wrap it so that we can store breadcrumbs on it.
+ * Get all breadcrumbs for the given span as well as it's parents.
  */
-function getActiveTransaction(): TransactionWithBreadcrumbs | undefined {
-  const activeSpan = getActiveSpan();
-  return activeSpan && (activeSpan.transaction as TransactionWithBreadcrumbs | undefined);
+function getBreadcrumbsForSpan(span: OtelSpan): Breadcrumb[] {
+  const events = span ? getOtelEvents(span) : [];
+
+  return events.map(otelEventToBreadcrumb);
+}
+
+function breadcrumbToOtelEvent(breadcrumb: Breadcrumb): Parameters<OtelSpan['addEvent']> {
+  const name = breadcrumb.message || '<no message>';
+
+  return [
+    name,
+    dropUndefinedKeys({
+      [OTEL_ATTR_BREADCRUMB_TYPE]: breadcrumb.type,
+      [OTEL_ATTR_BREADCRUMB_LEVEL]: breadcrumb.level,
+      [OTEL_ATTR_BREADCRUMB_EVENT_ID]: breadcrumb.event_id,
+      [OTEL_ATTR_BREADCRUMB_CATEGORY]: breadcrumb.category,
+      [OTEL_ATTR_BREADCRUMB_DATA]:
+        breadcrumb.data && Object.keys(breadcrumb.data).length > 0 ? JSON.stringify(breadcrumb.data) : undefined,
```

Comment:
l: Probably not too important for now but `JSON.stringify` fails if `breadcrumb.data` has circular references.

Reply:
I guess we have to serialize breadcrumbs here to save them on the span, right?

Reply:
yeah, data has to be a primitive (or array of primitives), sadly. I think in a follow up I'll try to store this in a flattened form via dot notation 🤔 so `'sentry.data.xx': 12` becomes `data: { xx: 12 }`. But that's probably a bit more expensive to serialize/deserialize 🤔

Reply:
Maybe we can use our `normalize` function to serialize but as I said, no need to worry about this now.

Reply:
true, will do this even for now, then we can improve this later further! 👍